### PR TITLE
Iam compact tech & socioeconomics harmonization

### PR DESCRIPTION
**Pull request by russellhz**

### CODE REVIEW

File: input/gcamdata/R/constants.R [hunk 1]
```diff
@@ -8,7 +8,7 @@ COMMENT_CHAR             <- "#"
 UNDER_TIMESHIFT          <- FALSE
 YEAR_PATTERN             <- "^(1|2)[0-9]{3}$"   # a 1 or 2 followed by three digits, and nothing else
 LOGIT_TYPE_COLNAME       <- "logit.type"        # will be removed by test code before old-new comparison
-DISABLED_MODULES         <- "NONE"
+DISABLED_MODULES         <- "policy"
```
Comment (jonsampedro) [inline input/gcamdata/R/constants.R:11]: DISABLED_MODULES should be "NONE", right?

File: input/gcamdata/R/zenergy_L210.resources.R [hunk 2]
```diff
@@ -537,8 +537,8 @@ module_energy_L210.resources <- function(command, ...) {
     L210.RsrcEnvironCost_SSP4 %>%
       # Set environmental costs for coal to 0 for low growth regions,
       # 10 * environcost for high growth regions
-      mutate(environCost = if_else(resource == "coal" & region %in% L210.low_reg, 0, environCost),
-             environCost = if_else(resource == "coal" & region %in% L210.high_reg, 10 * environCost, environCost)) %>%
+      mutate(environCost = dplyr::if_else(resource == "coal" & region %in% L210.low_reg, 0, input.cost),
```
Comment (jonsampedro) [inline input/gcamdata/R/zenergy_L210.resources.R:540]: Why do you need to change the environCost of the resources?

File: input/gcamdata/R/zenergy_xml_solar_adv.R [hunk 3]
```diff
@@ -27,31 +27,40 @@ module_energy_solar_adv_xml <- function(command, ...) {
...
-      add_xml_data(L223.GlobalTechCapital_sol_adv, "GlobalTechCapital") %>%
-      add_xml_data(L223.GlobalIntTechCapital_sol_adv, "GlobalIntTechCapital") %>%
-      add_xml_data(rooftop_pv_adv, "StubTechCost") %>%
-      add_precursors("L223.GlobalTechCapital_sol_adv", "L223.GlobalIntTechCapital_sol_adv",
-                     "L223.StubTechCapFactor_elec") ->
-      solar_adv.xml
+    # This xml is empty if using EUREF data
+    if (energy.ELEC_COST_SOURCE %in% c("EUREF", "WEO", "WEO-EUREF")){
+      create_xml("solar_adv.xml") ->
+        solar_adv.xml
```
Comment (jonsampedro) [inline input/gcamdata/R/zenergy_xml_solar_adv.R:33]: Does this create an empty xml? Don't you need a NULL object?

File: input/gcamdata/R/zenergy_xml_solar_low.R [hunk 4]
```diff
@@ -27,6 +27,11 @@ module_energy_solar_low_xml <- function(command, ...) {
     L223.GlobalIntTechCapital_sol_low <- get_data(all_data, "L223.GlobalIntTechCapital_sol_low")
     L223.StubTechCapFactor_elec <- get_data(all_data, "L223.StubTechCapFactor_elec")
 
+    # This xml is empty if using EUREF data
+    if (energy.ELEC_COST_SOURCE %in% c("EUREF", "WEO", "WEO-EUREF")){
+      create_xml("solar_low.xml") ->
+        solar_low.xml
```
Comment (jonsampedro) [inline input/gcamdata/R/zenergy_xml_solar_low.R:33]: Same comment!

File: input/harmonization/zzSOCIOECONOMICS_pop_gdp.R [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+  dplyr::mutate(value_MER = value*gdp_deflator(2010,2017)*MER/PPP) %>%
+  dplyr::group_by(region_GCAM3,year) %>%
+  dplyr::summarise(val = round(sum(value_MER, na.rm = TRUE)*gdp_deflator(1990,2010)/1e6))   # unit: million 1990 USD
+gdp = tidyr::pivot_wider(gdp, names_from = 'year', values_from = 'val')
+write.csv(gdp, file = "inst/extdata/socioeconomics/GCAM3_GDP_iamcompact.csv")
+
+
+
+########################################################
+check_gdp = read.csv('C:/Users/claudia.rodes/Documents/GitHub/lc-gcam-core/input/gcamdata/inst/extdata/socioeconomics/USDA_GDP_MER.csv', skip = 6)
```
Comment (jonsampedro) [inline input/harmonization/zzSOCIOECONOMICS_pop_gdp.R]: Change directory

File: input/gcamdata/R/constants.R (hunk 1 above)
Comment (russellhz) [reply input/gcamdata/R/constants.R:11]: Clàudia was getting an error with a policy chunk that I don't get when I run the datasystem. We will need to debug this eventually, not sure whether it's better to undo this disabling now

File: input/gcamdata/R/zenergy_L210.resources.R (hunk 2 above)
Comment (russellhz) [reply input/gcamdata/R/zenergy_L210.resources.R:540]: I think this is a bug in the code (nothing to do with our changes) that Clàudia had to fix. The column `environCost` doesn't actually exist yet in the data, so it throws an error with some package versions

File: input/gcamdata/R/zenergy_xml_solar_adv.R (hunk 3 above)
Comment (russellhz) [reply input/gcamdata/R/zenergy_xml_solar_adv.R:33]: this creates an empty xml for me. The `create_xml()` function here just produces a list with the filename and no other data.

**Comment by jonsampedro** [comment]
No more comments from my side (just looked at the code!)

File: input/harmonization/zzSOCIOECONOMICS_pop_gdp.R (hunk 5 above)
Comment (klau506) [reply input/harmonization/zzSOCIOECONOMICS_pop_gdp.R]: This file is no longer needed. I'll delete it

**Comment by klau506** [comment]
I've cleaned the repo from unnecessary files that I previously used for the GDP and population harmonization. Now, this harmonization is directly embedded in the DS

**Pull request merged**